add architecture section to migrate page, explaining why yugabyte (#53)
filecoin-project/boost-docs: tutorials/how-to-migrate-boost-v1-to-boost-v2.md

<figure><img src="../.gitbook/assets/Screenshot 2023-06-14 at 13.13.54.png" alt=""><figcaption></figcaption></figure>

### Architecture

When boost receives a storage deal, it creates an index of all the block
locations in the deal data, and stores the index in LID.

When boostd / booster-http etc gets a request for a block it:
- gets the block sector and offset from the LID index
- requests the data at that sector and offset from the miner

![LID Flow](./assets/LID-flow.png)

A large miner with many incoming retrieval requests needs many
boostd / booster-http / booster-bitswap processes to serve those requests.
These processes need to look up block locations in a centralized index.

> The boost team tested several databases and found that Yugabyte DB

Reviewer: ```suggestion
We tested several databases and found that YugabyteDB
```
